src/types.rs

```diff
@@ -919,8 +919,12 @@ new_secret_type![
 ];
 impl PartialEq for Nonce {
     fn eq(&self, other: &Self) -> bool {
-        constant_time::verify_slices_are_equal(self.secret().as_bytes(), other.secret().as_bytes())
-            .is_ok()
+        use subtle::ConstantTimeEq;
+        self.secret()
+            .as_bytes()
+            .ct_eq(other.secret().as_bytes())
+            .unwrap_u8()
```

Comment:
let's use `.into::<bool>()` or similar here as indicated in `subtle`'s [docs](https://docs.rs/subtle/2.4.1/subtle/struct.Choice.html#note)